pkg/routeagent/controllers/route/route_test.go

```diff
@@ -0,0 +1,46 @@
+package routecontroller
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("RouteController", func() {
+
+	Describe("Function populateCidrBlockList", func() {
+
+		Context("subnets does not contain CIDR blocks", func() {
+			It("should append CIDR block to subnets", func() {
+				routeController := RouteController{subnets: []string{"192.168.1.0/24"}}
+				routeController.populateCidrBlockList([]string{"10.10.10.0/24", "192.168.1.0/24"})
+				want := []string{"192.168.1.0/24", "10.10.10.0/24"}
+				Expect(routeController.subnets).To(Equal(want))
+			})
+		})
+
+		Context("subnets contain CIDR block", func() {
+			It("Should not append CIDR block subnets", func() {
+				routeController := RouteController{subnets: []string{"10.10.10.0/24", "192.168.1.0/24", "192.168.0.0./24"}}
+				routeController.populateCidrBlockList([]string{"10.10.10.0/24", "192.168.1.0/24"})
+				want := []string{"10.10.10.0/24", "192.168.1.0/24", "192.168.0.0./24"}
+				Expect(routeController.subnets).To(Equal(want))
+			})
+
+		})
+	})
+
+	Describe("Function containsString", func() {
+		Context("Given array of strings contain specified string", func() {
```

Comment:
How about "When the given string array contains the given string"

Reply:
2 times given seems confusing. "given" array of strings contains "specified" string is more clear as to what needs to be searched where. Meaning "specified" needs to be searched in "given".

Reply:
"given" and "specified" are synonymous in this case, ie both params are given or specified by the caller. I just used the same word  in both clauses for consistency but it's fine to mix them.